contracts/test-utils/transfer-wrapper/src/lib.rs

```diff
@@ -0,0 +1,262 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use dusk_abi::{ContractId, Transaction};
+use dusk_bytes::Serializable;
+use dusk_pki::{PublicSpendKey, SecretSpendKey, ViewKey};
+use lazy_static::lazy_static;
+use phoenix_core::{Crossover, Fee, Note};
+use rand::{CryptoRng, RngCore};
+use rusk_abi::RuskModule;
+use rusk_vm::{Contract, GasMeter, NetworkState, VMError};
+use transfer_circuits::ExecuteCircuit;
+use transfer_contract::{Call, Error as TransferError, TransferContract};
+
+use std::convert::{TryFrom, TryInto};
+
+use dusk_plonk::prelude::*;
+
+const TRANSFER: &'static [u8] = include_bytes!(
+    "../../../../target/wasm32-unknown-unknown/release/transfer_contract.wasm"
+);
+
+lazy_static! {
+    static ref PP: PublicParameters = unsafe {
+        let pp = rusk_profile::get_common_reference_string().unwrap();
+
+        PublicParameters::from_slice_unchecked(pp.as_slice())
+    };
+}
+
+pub fn transfer_contract() -> ContractId {
+    ContractId::from([
+        0xd3, 0xf8, 0x7f, 0xfc, 0x1b, 0xc7, 0x43, 0x1d, 0xde, 0x81, 0x5f, 0xb1,
+        0xe1, 0x1b, 0xd0, 0xfe, 0x88, 0x37, 0x1a, 0x15, 0x4a, 0xec, 0x27, 0x5d,
+        0xed, 0x2, 0x4d, 0x8c, 0xc0, 0xf7, 0x99, 0x5f,
+    ])
+}
+
+pub fn genesis<R>(
+    rng: &mut R,
+    block_height: u64,
+    balance: u64,
+) -> Result<(NetworkState, SecretSpendKey), TransferError>
+where
+    R: RngCore + CryptoRng,
+{
+    let mut network = NetworkState::with_block_height(block_height);
+
+    let ssk = SecretSpendKey::random(rng);
+    let psk = PublicSpendKey::from(&ssk);
+
+    let transfer = if balance > 0 {
+        let genesis = Note::transparent(rng, &psk, balance);
+
+        TransferContract::try_from(genesis)?
+    } else {
+        TransferContract::default()
+    };
+
+    let rusk_mod = RuskModule::new(&*PP);
+    network.register_host_module(rusk_mod);
+
+    let transfer = Contract::new(transfer, TRANSFER.to_vec());
+    let transfer = network
+        .deploy(transfer)
+        .map_err(|_| TransferError::ContractNotFound)?;
+
+    assert_eq!(transfer, transfer_contract());
+
+    Ok((network, ssk))
+}
+
+pub fn transfer_state(
+    network: &NetworkState,
+) -> Result<TransferContract, VMError> {
+    network.get_contract_cast_state(&transfer_contract())
+}
+
+fn circuit_key(circuit_id: &[u8; 32]) -> Result<ProverKey, VMError> {
+    let pk = rusk_profile::keys_for(circuit_id)
+        .and_then(|keys| keys.get_prover())
+        .map_err(|e| VMError::ContractPanic(e.to_string()))?;
+    let pk = ProverKey::from_slice(pk.as_slice())
+        .map_err(|e| VMError::ContractPanic(e.to_string()))?;
+
+    Ok(pk)
+}
+
+fn prepare_execute<'a, R, I>(
```

Comment:
Add documentation: What this function does?

Reply:
This is private scope, but I added a comment